### ISSUE TITLE
Unable to use Play 2.4.0-M3 cache 

### ISSUE THREAD

**Issue by ulejon** [scaldi/scaldi-play#13]
I'm trying to use the cache api in my application. However, when trying to use the cache in one of my controllers I get the following stacktrace:

```
[error] application -

! @6lpffabbl - Internal server error, for (GET) [/] ->

play.api.http.HttpErrorHandlerExceptions$$anon$1: Execution exception[[NullPointerException: null]]
	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:254) ~[play_2.11-2.4.0-M3.jar:2.4.0-M3]
	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:180) ~[play_2.11-2.4.0-M3.jar:2.4.0-M3]
	at play.core.server.Server$class.logExceptionAndGetResult$1(Server.scala:50) [play-server_2.11-2.4.0-M3.jar:2.4.0-M3]
	at play.core.server.Server$$anonfun$getHandlerFor$4.apply(Server.scala:59) [play-server_2.11-2.4.0-M3.jar:2.4.0-M3]
	at play.core.server.Server$$anonfun$getHandlerFor$4.apply(Server.scala:57) [play-server_2.11-2.4.0-M3.jar:2.4.0-M3]
Caused by: java.lang.NullPointerException: null
	at scaldi.util.ReflectionHelper$.isAssignableFrom(ReflectionHelper.scala:72) ~[scaldi_2.11-0.5.4.jar:0.5.4]
	at scaldi.TypeTagIdentifier.sameAs(Identifier.scala:63) ~[scaldi_2.11-0.5.4.jar:0.5.4]
	at scaldi.Identifier$$anonfun$1$$anonfun$apply$1.apply(Identifier.scala:19) ~[scaldi_2.11-0.5.4.jar:0.5.4]
	at scaldi.Identifier$$anonfun$1$$anonfun$apply$1.apply(Identifier.scala:19) ~[scaldi_2.11-0.5.4.jar:0.5.4]
	at scala.collection.TraversableLike$$anonfun$filterImpl$1.apply(TraversableLike.scala:259) ~[scala-library-2.11.6.jar:na]
```

Some information about my project:

Play Version: 2.4.0-M3
Scaldi-play version:  0.5-play-2.4.0-M3-4
sbt version: 0.13.8
scala version: 2.11.6

application.conf:
```
play.application.loader = scaldi.play.ScaldiApplicationLoader

play.modules.enabled += "modules.UserModule"
play.modules.enabled += "scaldi.play.ControllerInjector"
```

UserModule:
```
package modules
import controllers.Application
import scaldi.Module

class UserModule extends Module {
    bind [Application] to new Application
}
```

Application:
```
package controllers

import play.api.cache.CacheApi
import play.api.mvc._
import scaldi.{Injectable, Injector}

class Application(implicit inj: Injector) extends Controller with Injectable {
  val cache = inject [CacheApi]

  def index = Action { implicit request =>
    val cachedValue = cache.getOrElse("someProp"){
      "new prop"
    }
    Ok(views.html.index(s"$cachedValue"))
  }
}
```

routes:
```
GET        /assets/*file            controllers.Assets.at(path="/public", file)
GET        /                        @controllers.Application.index()
```

What am I missing?

**Comment by OlegIlyenko** [comment]
Thanks for reporting this issue - it's a bug.  I fixed it and will promote version **0.5-play-2.4.0-M3-5** as soon as possible (looks like there are some latency issues at the maven central at the moment).

In general the code you have shown looks good and correct. One small note though: You are binding your `Application` controller manually like this:

```
class UserModule extends Module {
    bind [Application] to new Application
}
```

and then you also adding `ControllerInjector` in the `application.conf`:

    play.modules.enabled += "scaldi.play.ControllerInjector"

It is not necessary in this case - you need to either do it manually or use `ControllerInjector` which will add all of the controllers automatically.

**Comment by ulejon** [comment]
Hi! 

Thanks for the quick feedback. 
Yeah, I suspected the manual binding of the controller wasn't necessary. After a while trying to figure out if the bug was in my code or not I was desperate and tried all kinds of hacks :-)

I'll keep an eye open for the next version. Great job!
